src/main/java/lotto/controller/GameController.java

```diff
@@ -0,0 +1,104 @@
+package lotto.controller;
+
+import lotto.service.GameService;
+import lotto.service.LottoService;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class GameController {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final GameService gameService;
+    private final LottoService lottoService;
+
+    public GameController(InputView inputView, OutputView outputView, GameService gameService, LottoService lottoService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.gameService = gameService;
+        this.lottoService = lottoService;
+    }
+
+    public void run() {
+        purchaseLottos();
+        issueLottos();
+        inputWinningLotto();
+        inputBonusNumber();
+        outputWinningStatistics();
+        outputProfitRate();
+    }
+
+    public void purchaseLottos() {
+        String input;
+        do {
+            outputView.requestPurchaseAmount();
+            input = inputView.scanPurchaseAmount();
+        } while (isInvalidPurchaseAmount(input));
+    }
```

Comment:
java 컨벤션에서 do while 을 지양하고 있습니다! 그리고 해당 코드를 이렇게 바꿔보는건 어떨까요?


```suggestion
    public void purchaseLottos() {
        try {
            outputView.requestPurchaseAmount();
            String input = inputView.scanPurchaseAmount();
            gameService.initPurchaseAmount(input);
        } catch (IllegalArgumentException e) {
            outputView.printException(e.getMessage());
            purchaseLottos();
        }
    }
```

이렇게 수정하면 아래의 isInvalidPurchaseAmount(String input) 함수도 없앨 수 있을 것 같아요

Reply:
재귀의 형식으로 반복 로직을 구현할 수 있다는 사실을 전혀 생각하지 못했던 것 같습니다! 덕분에 좋은 인사이트 얻어갑니다. 감사합니다 :)